client/components/forms/form-radios-bar/index.jsx

```diff
@@ -0,0 +1,45 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+
+import FormLabel from 'components/forms/form-label';
+import FormRadio from 'components/forms/form-radio';
+import FormRadioWithThumbnail from 'components/forms/form-radio-with-thumbnail';
+
+const FormRadiosBar = ( { isThumbnail, checked, onChange, items } ) => {
+	const addProps = ( Component, item, key ) =>
+		<Component key={ key } checked={ checked === item.value } onChange={ onChange } { ...item } />;
+
+	return (
+		<div className={ classnames( 'form-radios-bar', { 'is-thumbnail': isThumbnail } ) }>
+			{ items.map(
+				( item, i ) =>
+					isThumbnail
+						? addProps( FormRadioWithThumbnail, item, i )
+						: <FormLabel key={ i }>
```

Comment:
Oops on the index, small oversight on my end! Usually the `item.value` should be sufficient given the exclusionary selection nature of radio buttons. But maybe just to be absolutely sure, we go with `item.value + i`. ;)

As for the parenthesis around the ternary, those, unfortunately, get stripped out when using Calypso prettier. Maybe something to suggest as a change for the sake of readability.